osu.Game/Overlays/Profile/Sections/Kudosu/DrawableKudosuHistoryItem.cs

```diff
@@ -0,0 +1,150 @@
+                {
+                    Anchor = Anchor.CentreRight,
+                    Origin = Anchor.CentreRight,
+                }
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            date.Colour = colours.GreySeafoamLighter;
+
+            string userLinkTemplate() => $"[{historyItem.Giver?.Url} {historyItem.Giver?.Username}]";
+
+            switch (historyItem.Action)
+            {
+                case KudosuAction.VoteGive:
+                    linkFlowContainer.AddText(@"Received ");
+                    addKudosuPart();
+                    addMainPart(@" from obtaining votes in modding post of ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.Give:
+                    linkFlowContainer.AddText(@"Received ");
+                    addKudosuPart();
+                    addMainPart($@" from {userLinkTemplate()} for a post at ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.Reset:
+                    addMainPart($@"Kudosu reset by {userLinkTemplate()} for the post ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.VoteReset:
+                    linkFlowContainer.AddText(@"Lost ");
+                    addKudosuPart();
+                    addMainPart(@" from losing votes in modding post of ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.DenyKudosuReset:
+                    linkFlowContainer.AddText(@"Denied ");
+                    addKudosuPart();
+                    addMainPart(@" from modding post ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.Revoke:
+                    addMainPart($@"Denied kudosu by {userLinkTemplate()} for the post ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.AllowKudosuGive:
+                    linkFlowContainer.AddText(@"Received ");
+                    addKudosuPart();
+                    addMainPart(@" from kudosu deny repeal of modding post ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.DeleteReset:
+                    linkFlowContainer.AddText(@"Lost ");
+                    addKudosuPart();
+                    addMainPart(@" from modding post deletion of ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.RestoreGive:
+                    linkFlowContainer.AddText(@"Received ");
+                    addKudosuPart();
+                    addMainPart(@" from modding post restoration of ");
+                    addPostPart();
+                    break;
+
+                case KudosuAction.RecalculateGive:
+                    linkFlowContainer.AddText(@"Received ");
```

Comment:
these could be moved/grouped to a switch statement within `addKudosuPart` (which could them be called regardless of the action, returning in `default` if not requried).